ansible/roles/piwik/templates/nginx-site-piwik.j2

```diff
@@ -0,0 +1,67 @@
+server {
+    server_name {{ server_name}};
+
+    listen 443 ssl default_server;
+    listen [::]:443 ssl default_server;
+
+    ssl_certificate /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/privkey.pem;
+    ssl_trusted_certificate /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/fullchain.pem;
+
+  if ($request_method !~ ^(GET|HEAD|POST)$ ) {
+    return 444;
+  }
+
+  root  {{ app_path }}/current/piwik;
+  index  index.php index.html;
+  # Disallow any usage of piwik assets if referer is non valid.
+  location ~* ^.+\.(?:jpg|png|css|gif|jpeg|js|swf)$ {
+    # Defining the valid referers.
+    #valid_referers none blocked *.mysite.com othersite.com;
+
+    expires max;
```

Comment:
I don't think this is an issue, because we don't use the image based tracking of piwik, but only the js based one.

Since E-Tag is set and the piwik JS script is fetched via javascript the caching rule should be good. If for some reason there is the need to update the software I believe E-Tag should be updated and the browser cache will be busted. If we need to force busting of the cache we can always do something like `/piwik.js?$RANDOM_NONCE`.